### ISSUE TITLE
Improve code of SFTP back up flow

### ISSUE THREAD

**Issue by mmorhun** [eclipse/che#20836]
### Is your task related to a problem? Please describe

Current flow is not optimal. We should improve it.

### Describe the solution you'd like

Refactor code that is responsible for SFTP backup flow

**Comment by mmorhun** [comment]
When I started looking into this issue, I found that SFTP backup mechanism is broken.
Short investigation showed that there are 3 problems (all is caused by random user ID in the Che Operator container):
1. User home is not writable (defined as `/`), so it fails to update ssh config in `~/.ssh` folder. A quick fix is to add the following line in Dockerfile: ```env HOME=/tmp/```
2. No ssh client in UBI image. To fix, just install `openssh-clients` package using `microdnf install -y openssh-clients` or add the package to already existing install command
3. ssh fails to open connection with the following error: `No user exists for uid xxxx`. This cause by absent record in passswd file. To fix, add write permissions for the `/etc/passwd` and add current user record in entrypoint script.
